From 82049e9b3399d1d93b46e68fc2a5fcdf71ef9d60 Mon Sep 17 00:00:00 2001
From: Mickael Reulier <mickael.reulier@st.com>
Date: Wed, 30 Aug 2017 11:23:34 +0200
Subject: [PATCH 20/25] glimagesink: don't use wl_egl_window as a proxy object

There is a crash in gcnano when setting proxy queue of
window.native (due to memory corruption).
For gcnano, wl_egl_window is not a wl_proxy object,
we can not set queue to wl_egl_window object.

Buglink: https://stintbugzilla.st.com/show_bug.cgi?id=111353
Change-Id: I739e029d043fe1d988bfcd0e9b90d656cf691d3f
Signed-off-by: Mickael Reulier <mickael.reulier@st.com>
Reviewed-on: https://gerrit.st.com/76789
Reviewed-by: CITOOLS <smet-aci-reviews@lists.codex.cro.st.com>
Reviewed-by: Vincent ABRIOU <vincent.abriou@st.com>

%% original patch: 0024-glimagesink-don-t-use-wl_egl_window-as-a-proxy-objec.patch

%% original patch: 0024-glimagesink-don-t-use-wl_egl_window-as-a-proxy-objec.patch

diff --git a/gst-libs/gst/gl/wayland/gstglwindow_wayland_egl.c b/gst-libs/gst/gl/wayland/gstglwindow_wayland_egl.c
index 933ddfe..cc6a543 100644
--- a/gst-libs/gst/gl/wayland/gstglwindow_wayland_egl.c
+++ b/gst-libs/gst/gl/wayland/gstglwindow_wayland_egl.c
@@ -319,9 +319,6 @@ create_surfaces (GstGLWindowWaylandEGL * window_egl)
 
     window_egl->window.native =
         wl_egl_window_create (window_egl->window.surface, width, height);
-    if (window_egl->window.queue)
-      wl_proxy_set_queue ((struct wl_proxy *) window_egl->window.native,
-          window_egl->window.queue);
   }
 }
 
-- 
2.7.4

